src/gui/ogr/qgsvectorlayersaveasdialog.cpp

```diff
@@ -671,90 +665,181 @@ void QgsVectorLayerSaveAsDialog::mFormatComboBox_currentIndexChanged( int idx )
+      if ( mAttributeTable->isColumnHidden( static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) ) ||
+           ! mAttributeTable->item( row, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) ) ||
+           !( mAttributeTable->item( row, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) )->flags() & Qt::ItemIsUserCheckable ) )
+        return;
+
+      if ( mAttributeTable->item( row, column )->checkState() == Qt::Unchecked )
       {
-        checkBoxEnabled = true;
-        break;
+        mAttributeTable->item( row, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) )->setCheckState( Qt::Unchecked );
+        mAttributeTable->item( row, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) )->setFlags( Qt::ItemIsUserCheckable );
+        bool checkBoxEnabled = false;
+        for ( int i = 0; i < mAttributeTable->rowCount(); i++ )
+        {
+          if ( mAttributeTable->item( i, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) ) &&
+               mAttributeTable->item( i, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) )->flags() & Qt::ItemIsEnabled )
+          {
+            checkBoxEnabled = true;
+            break;
+          }
+        }
+        mReplaceRawFieldValues->setEnabled( checkBoxEnabled );
+        if ( !checkBoxEnabled )
+          mReplaceRawFieldValues->setCheckState( Qt::Unchecked );
+      }
+      else if ( mAttributeTable->item( row, column )->checkState() == Qt::Checked )
+      {
+        mAttributeTable->item( row, static_cast<int>( ColumnIndex::ExportAsDisplayedValue ) )->setFlags( Qt::ItemIsUserCheckable | Qt::ItemIsEnabled );
+        mReplaceRawFieldValues->setEnabled( true );
       }
     }
-    mReplaceRawFieldValues->setEnabled( checkBoxEnabled );
-    if ( !checkBoxEnabled )
-      mReplaceRawFieldValues->setCheckState( Qt::Unchecked );
-  }
-  else if ( column == COLUMN_IDX_NAME &&
-            mAttributeTable->item( row, column )->checkState() == Qt::Checked &&
-            mAttributeTable->columnCount() == 3 &&
-            mAttributeTable->item( row, COLUMN_IDX_EXPORT_AS_DISPLAYED_VALUE ) &&
-            ( mAttributeTable->item( row, COLUMN_IDX_EXPORT_AS_DISPLAYED_VALUE )->flags() & Qt::ItemIsUserCheckable ) )
-  {
-    mAttributeTable->item( row, COLUMN_IDX_EXPORT_AS_DISPLAYED_VALUE )->setFlags( Qt::ItemIsUserCheckable | Qt::ItemIsEnabled );
-    mReplaceRawFieldValues->setEnabled( true );
-  }
-  else if ( column == COLUMN_IDX_EXPORT_AS_DISPLAYED_VALUE &&
-            ( mAttributeTable->item( row, column )->flags() & Qt::ItemIsUserCheckable ) )
-  {
-    bool allChecked = true;
-    bool allUnchecked = true;
-    for ( int i = 0; i < mAttributeTable->rowCount(); i++ )
+    break;
+    case ColumnIndex::ExportName:
+    {
+      // Rename eventually duplicated names
+      QStringList names = attributesExportNames();
+      while ( names.count( item->text() ) > 1 )
+        item->setText( QString( "%1_2" ).arg( item->text() ) );
```

Comment:
This logic will not handle a triple collision nor all collisions with the resolved new name.

Reply:
then we should also check for empty strings.
another approach would be to prevent clicking until names are correctly set, it might make more sense.

Reply:
An empty string can be replaced with the normal field name or its alias, depending on the checkbox's state

Reply:
Whoops! You're absolutely right, sorry about that!